### ISSUE TITLE
Only for DPX?

### ISSUE THREAD

**Issue by melMass** [videovillage/oiiococoa#4]
Hi,

Thanks for oiiococoa, I'm making a swift library to handle NSImage's through OIIO. I know both C++ and C quite well, but I'm a bit lost regarding ObjC. Reading the code quickly, it seems that most methods are tailored specifically for dpx?

I'm basically looking to do almost the same thing with EXR images and wanted to know if this library was supporting it.

Thanks

**Comment by gregcotten** [comment]
Hey Mel,

MacOS + iOS have decent support for EXR out of the box for both NSImage and CGImage - is there a reason you need to use olio to do that?

**Comment by melMass** [comment]
Thanks!
The main idea would be to support OCIO configs and Colorspace management of float images.
I managed to make a very basic swift wrapper of your library as is (I'm going to upload the test repo just after answering) , it builds but cannot open the EXR thought OIIO.

- This Bypasses OIIO and load the image using the default backend  
  ```swift
  let image = NSImage.oiio_image(withContentsOf: url)
  ```
- So I tried the force method
  ```swift
  let image = NSImage.oiio_force(withContentsOf: url)
  ```
  but this creates the following error:
  ```ruby
  CGImageCreate: invalid image bits/component: 16 bits/pixel 80 alpha info = kCGImageAlphaLast
  ERROR: CGImageDestinationAddImage: cgImage is nil
  finalize:2670: image destination must have at least one image
  ```
- And these always returns `nil` (to be expected as the loading fail):
  ![image](https://user-images.githubusercontent.com/7041726/128331437-6cadc014-9409-41ab-ba51-d56d24b35369.png)
